src/screens/FeedScreen/Header.js

```diff
@@ -102,15 +114,41 @@ const _renderAnonimity = ({
   height,
   headerStyle,
   showAnonymousOption = false,
-  onHeaderOptionClicked = () => {},
-  hideThreeDot,
   version = 1,
   anonUserInfo = {},
   isPostDetail,
   karmaScore = 0,
-  isBlurredPost = false
+  isBlurredPost = false,
+  isFollow = false,
+  onPressFollUnFoll = () => {},
+  onDeletePost = () => {},
+  isShowDelete = false,
+  isSelf = false,
+  hideThreeDot,
+  onHeaderOptionClicked = () => {}
 }) => {
   const navigation = useNavigation();
+  const refSheet = React.useRef();
+  const dataSheet = [
+    {
+      id: 1,
+      name: 'Share link',
+      icon: <ShareAndroidIcon />
+    }
+  ];
+
+  if (isShowDelete) {
+    dataSheet.push({
+      id: 2,
+      name: 'Delete post',
+      icon: <TrashRed />,
+      onPress: () => {
+        refSheet.current.close();
+        onDeletePost();
+      },
+      style: {color: colors.red}
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isShowDelete` prop is used to conditionally add a delete option to the `dataSheet` array. This is a good practice as it allows for dynamic rendering of options based on the context. However, I would suggest moving this logic outside of the component function to avoid unnecessary computations during each render. You can use a `useMemo` hook to achieve this.

```diff
-  const dataSheet = [
-    {
-      id: 1,
-      name: 'Share link',
-      icon: <ShareAndroidIcon />
-    }
-  ];
-
-  if (isShowDelete) {
-    dataSheet.push({
-      id: 2,
-      name: 'Delete post',
-      icon: <TrashRed />,
-      onPress: () => {
-        refSheet.current.close();
-        onDeletePost();
-      },
-      style: {color: colors.red}
-    });
-  }

+  const dataSheet = React.useMemo(() => {
+    const baseDataSheet = [
+      {
+        id: 1,
+        name: 'Share link',
+        icon: <ShareAndroidIcon />
+      }
+    ];
+
+    if (isShowDelete) {
+      baseDataSheet.push({
+        id: 2,
+        name: 'Delete post',
+        icon: <TrashRed />,
+        onPress: () => {
+          refSheet.current.close();
+          onDeletePost();
+        },
+        style: {color: colors.red}
+      });
+    }
+
+    return baseDataSheet;
+  }, [isShowDelete, onDeletePost]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->